### ISSUE TITLE
LinkLuaAbility

### ISSUE THREAD

**Issue by Perryvw** [ModDota/BugTracker#94]
We have the `LinkLuaModifier` function in the Lua API which is useful. I am requesting something similar for Lua abilities, since most lua abilities require an entry in npc_abilities_custom.txt containing just the name, baseclass and scriptfile. Having the power to link lua abilities without needing to define anything outside of Lua would allow for more self-contained lua and cleaner abilities_custom files.

**Comment by Arhowk** [comment]
Isn't this physically impossible because default dota abilities use the npc_abilities file? There's no definitions of modifiers (besides a list of their names) in the exposed DotA files which allows for LinkLuaModifier behavior, so I don't see this as being feasible barring a rework of some internal systems.

**Comment by Perryvw** [comment]
That would depend on the moment the abilities are actually required, on the definition of units or on creation. If it is on creation you could call LinkLuaAbility before that moment. LinkLuaModifier works in a similar fashion; you can define lua modifiers from your addon_game_mode and then apply them from your datadriven ability.

**Comment by dustingraham** [comment]
I agree it should be very feasible in theory. LinkLuaAbility seems like a reasonable function to provide.
